src/options.ts

```diff
@@ -30,7 +30,7 @@ export enum ServiceOption {
 
 export type Options = {
   context: boolean;
-  snakeToCamel: boolean;
+  snakeToCamel: boolean | string;
```

Comment:
Ah yeah; this makes sense, although what might be nifty is to make this:

```typescript
  snakeToCamel: Array<'json' | 'keys'>;
```

So that the call sites like `maybeSnakeToCamel` and do just `options.snakeToCamel.includes('keys')` and then inside of `optionsFromParamter` we can handle "boolean --> string[]" kinda like:

```typescript
  if ((options.snakeToCamel as any) === false) {
    options.snakeToCamel = [];
  } else if ((options.snakeToCamel as any) === true) {
    options.snakeToCamel = ['keys', 'json'];
  }
```

Which has a few `as any`s which is a little weird, but we've done it for a few other options like this have that "might be boolean/might be strings".

Maybe try that? I think it will simplify the call sites quite a bit.

Reply:
@ajaykarthikr I'm going to go ahead and merge this and do ^ as a follow up. Thanks for the PR!